pkg/apicast/apicast.go

```diff
@@ -740,7 +741,14 @@ func (a *APIcast) computeHashedSecretData(ctx context.Context, k8sclient client.
 func HashSecret(data map[string][]byte) string {
 	hash := sha256.New()
 
-	for key, value := range data {
+	sortedKeys := make([]string, 0, len(data))
+	for k := range data {
+		sortedKeys = append(sortedKeys, k)
+	}
+	sort.Strings(sortedKeys)
+
+	for _, key := range sortedKeys {
+		value := data[key]
 		combinedKeyValue := append([]byte(key), value...)
 		hash.Write(combinedKeyValue)
 	}
```

Comment:
@carlkyrillos It looks good, but I'd like to ask - what is the reason that key and value are treated as a single entity when hashing?
Another approach - distinct entities:
for _, key := range sortedKeys {
	hash.Write([]byte(key))
	hash.Write(data[key])
}
Thanks

Reply:
I combined the key and value so that the hash will change if either the key _or_ the value changes because either should trigger an update.

Reply:
hey @carlkyrillos , seems to me that in my example hash will also change if either the key or the value changes. But doesn't matter, if I will have time, I will check, but later. Thank you